### PR TITLE
feat: add host control

### PR DESCRIPTION
**Pull request by moritzkalwa**
Added the host control system, that designates a host and gives him the ability to control what other participants can do.

### CODE REVIEW

**Comment by moritzkalwa** [comment]
Also added looping and fixed the frontend port in the fork.

File: backend/room.ts [hunk 1]
```diff
@@ -89,25 +131,35 @@ class Room {
...
+          id,
+        })),
         membersLoading: this.membersLoading,
         queue: await Promise.all(this.queue),
       }
     })()
   }
 
   getMember = (id: string) => this.members.find(m => m.client.id === id)
+  getId = (name: string) => this.members.find(m => m.name === name)?.client.id
```
Comment (vaaski) [inline backend/room.ts:150]: this is unused, right?
names aren't unique and shouldn't be used to identify users.

Comment (moritzkalwa) [reply backend/room.ts:149]: yep, this is in fact unused. stems from when the permissions used names to identify users.

File: src/util.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ interface LocalStored {
   "resync-volume": number
   "resync-muted": boolean
   "resync-last-room": string
-  "secret": string
+  secret: string
```
Comment (moritzkalwa) [inline src/util.ts:8]: wait, why is this diefferent than the other local storage values?

Comment (vaaski) [reply src/util.ts:8]: the quotes are only required when the property name includes special characters like `-`.
but yea i'll change it to resync-secret in the next commit.

**Comment by vaaski** [comment]
I have identified 3 issues so far:
- when reassigning the host after they left, the new host ends up with permission level `1` instead of `7`
- the actual syncing part seems to be broken when joining a room with a running video
- when looping is disabled the source doesn't clear anymore

**Comment by moritzkalwa** [comment]
> * [ ]  the actual syncing part seems to be broken when joining a room with a running video
> * [ ]  when looping is disabled the source doesn't clear anymore

please clarify these two

**Review by vaaski — APPROVED** [review]

**Pull request merged**